packages/relay/src/lib/relay.ts

```diff
@@ -137,12 +137,35 @@ export class RelayImpl implements Relay {
     const chainId = prepend0x(Number(configuredChainId).toString(16));
 
     const duration = constants.HBAR_RATE_LIMIT_DURATION;
-    const total = constants.HBAR_RATE_LIMIT_TOTAL.toNumber();
-    const hbarLimiter = new HbarLimit(logger.child({ name: 'hbar-rate-limit' }), Date.now(), total, duration, register);
+    const total = constants.HBAR_RATE_LIMIT_TOTAL;
 
     this.eventEmitter = new EventEmitter();
     this.cacheService = new CacheService(logger.child({ name: 'cache-service' }), register);
-    const hapiService = new HAPIService(logger, register, hbarLimiter, this.cacheService, this.eventEmitter);
+
+    const hbarSpendingPlanRepository = new HbarSpendingPlanRepository(
+      this.cacheService,
+      logger.child({ name: 'hbar-spending-plan-repository' }),
```

Comment:
@Nana-EC hbar-spending-plan doesnt sound very descriptive to me :? i think it may be better to keep the repository , its more descriptive

Reply:
Addressed but left repository in there

Reply:
Problem is it makes the logs longer and larger. This can be problematic for some systems.
In general it's easy to see a log and find the code associated with it.
I'm not even sure it's necessary to have a different name for the spending plans.
We have been using it so far to break out really large modules e.g. relay vs server vs ws server etc.
I won't block but let's think on this

Reply:
Noting to think about it :)

Reply:
Opened issue #3134 to address this and move forward with the merge into main.